File: qc-core/src/main/java/com/eager/questioncloud/core/domain/point/repository/ChargePointPaymentRepository.java
package com.eager.questioncloud.core.domain.point.repository;

import com.eager.questioncloud.core.common.PagingInformation;
import com.eager.questioncloud.core.domain.point.model.ChargePointPayment;
import java.util.List;

public interface ChargePointPaymentRepository {
    ChargePointPayment save(ChargePointPayment chargePointPayment);

    Boolean isCompletedPayment(Long userId, String paymentId);

    ChargePointPayment getChargePointPaymentForApprove(String paymentId);

    ChargePointPayment findByPaymentId(String paymentId);

    Boolean existsByPaymentId(String paymentId);

    List<ChargePointPayment> getChargePointPayments(Long userId, PagingInformation pagingInformation);

    int countByUserId(Long userId);
}

File: qc-core/qc-point/qc-point-core/src/main/kotlin/com/eager/questioncloud/point/repository/ChargePointPaymentRepository.kt
package com.eager.questioncloud.point.repository

import com.eager.questioncloud.common.pagination.PagingInformation
import com.eager.questioncloud.point.domain.ChargePointPayment

interface ChargePointPaymentRepository {
    fun save(chargePointPayment: ChargePointPayment): ChargePointPayment
    
    fun update(chargePointPayment: ChargePointPayment): ChargePointPayment
    
    fun isCompletedPayment(userId: Long, orderId: String): Boolean
    
    fun findByOrderIdWithLock(orderId: String): ChargePointPayment
    
    fun findByOrderId(orderId: String): ChargePointPayment
    
    fun getChargePointPayments(
        userId: Long,
        pagingInformation: PagingInformation
    ): List<ChargePointPayment>
    
    fun countByUserId(userId: Long): Int
    
    fun getPendingPayments(): List<ChargePointPayment>
    
    fun deleteAllInBatch()
}
